etl/src/v2/replication/apply.rs

```diff
@@ -159,39 +190,54 @@ struct ApplyLoopState {
     /// of restarts and allows Postgres to determine whether some old entries could be pruned from the
     /// WAL.
     last_commit_end_lsn: Option<PgLsn>,
+
     /// The LSN of the commit WAL entry of the transaction that is currently being processed.
     ///
     /// This LSN is set at every `BEGIN` of a new transaction, and it's used to know the `commit_lsn`
     /// of the transaction which is currently being processed.
     remote_final_lsn: Option<PgLsn>,
+
     /// The LSNs of the status update that we want to send to Postgres.
     next_status_update: StatusUpdate,
-    /// An enum indicating whether the processing of a batch should be terminated early, with a consequent
-    /// termination of the main apply loop too.
-    early_break: Option<BatchEarlyBreak>,
+
+    /// Maximum time to wait for a batch to fill
+    max_batch_fill_duration: Duration,
```

Comment:
nit: this is a configuration parameter which should not belong to the state, since it's not meant to change.

Reply:
Great catch! moved this outside the struct.